### PR TITLE
ignore native broker initialize check if application is not top-frame

### PR DESCRIPTION
**Pull request by lalimasharda**

### CODE REVIEW

File: lib/msal-browser/src/controllers/StandardController.ts [hunk 1]
```diff
@@ -1299,11 +1299,13 @@ export class StandardController implements IController {
         // Block redirectUri opened in a popup from calling MSAL APIs
         BrowserUtils.blockAcquireTokenInPopups();
 
-        // Block token acquisition before initialize has been called if native brokering is enabled
-        BrowserUtils.blockNativeBrokerCalledBeforeInitialized(
+        // Block token acquisition before initialize has been called if native brokering is enabled in top-frame
+        if (setInteractionInProgress) {
+            BrowserUtils.blockNativeBrokerCalledBeforeInitialized(
             this.config.system.allowNativeBroker,
```
Comment (peterzenz) [inline lib/msal-browser/src/controllers/StandardController.ts]: Nit:  Indent this line and the next

**Review by peterzenz — APPROVED** [review]
Note that small changes like this should have a test covering this feature explicitly.  As you look at the test automation plan, please keep this in mind.

File: lib/msal-browser/src/controllers/StandardController.ts (hunk 1 above)
Comment (lalimasharda) [reply lib/msal-browser/src/controllers/StandardController.ts]: Fixed!

File: lib/msal-browser/src/controllers/StandardController.ts [hunk 2]
```diff
@@ -1299,11 +1299,13 @@ export class StandardController implements IController {
         // Block redirectUri opened in a popup from calling MSAL APIs
         BrowserUtils.blockAcquireTokenInPopups();
 
-        // Block token acquisition before initialize has been called if native brokering is enabled
-        BrowserUtils.blockNativeBrokerCalledBeforeInitialized(
-            this.config.system.allowNativeBroker,
-            this.initialized
-        );
+        // Block token acquisition before initialize has been called if native brokering is enabled in top-frame
+        if (setInteractionInProgress) {
```
Comment (sameerag) [inline lib/msal-browser/src/controllers/StandardController.ts]: The check has always been `interactionInProgress` to identify an embedded frame I presume?

**Review by sameerag — APPROVED** [review]

File: lib/msal-browser/src/controllers/StandardController.ts (hunk 2 above)
Comment (lalimasharda) [reply lib/msal-browser/src/controllers/StandardController.ts]: Yes, I just need to change the naming.

**Review by sameerag — APPROVED** [review]
Makes it much clear. Thanks.

**Review by tnorling — APPROVED** [review]

**Pull request merged**